OPHD/States/CrimeExecution.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "../UI/NotificationArea.h"
+#include "../Things/Structures/Structures.h"
+#include <vector>
+
+
+class CrimeExecution 
+{
+public:
+	CrimeExecution(NotificationArea& notificationArea);
+
+	void executeCrimes(std::vector<Structure*> structuresCommittingCrime);
+	void executeCrimes(std::vector<Structure*>&& structuresCommittingCrime);
```

Comment:
Is there any plan to update or store the contents of the passed parameter? If not, I would recommend a single overload passing by `const&`. A `const&` parameter binds to both lvalues and rvalues. Looking at the code, it appears read-only access is sufficient.

If you need write access to the object, and want to be able to accept both named variables and temporary objects, then it makes sense to have both `&` and `&&` overloads.

Here it's by-value and rvalue reference. So write access would be granted to temporaries, but named variables would be copied. I suspect this was not the intention.

Reply:
Yeah, this should be const& as you point out. I can update tomorrow.